x/tokenfactory/types/denoms.go

```diff
@@ -0,0 +1,46 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const (
+	ModuleDenomPrefix = "factory"
+)
+
+func GetTokenDenom(creator, nonce string) (string, error) {
+	denom := strings.Join([]string{ModuleDenomPrefix, creator, nonce}, "/")
+	return denom, sdk.ValidateDenom(denom)
+}
```

Comment:
I think we should have a safety assert somewhere that `creator` has no slashes in it. Otherwise there is a malleability attack between creator and nonce.